### DISCUSSION TITLE
How do you put a report together?

### DISCUSSION THREAD

**Discussion by jeffmarshall** [manusimidt/py-xbrl#55] [Q&A] [answered]
I have the XBRL for Microsoft's latest 10-K filing, and I would like to reconstruct the income statement, balance sheet, and cash flow statements, but I have no idea how to navigate the XBRL instance (`<xbrl.instance.XbrlInstance at 0x108782850>`). How might I start? I have looked at the XML documents available with the filing, and it looks like the `FilingSummary.xml` and the `.xsd` file have information about the reports, and I believe the `_pre.xml` file would explain where everything is meant to go, but how do I make sense of it in code? Any tips would be appreciated 😄

**Comment by manusimidt** [comment, marked as answer]
Hello, 

If you have a presentation linkbase avaliable, you can reconstruct the structure of the income statement, balance sheet and cash flow statement. The parser also parses the presentation linkbase, Unfortunately, it is still relatively cumbersome to work with it.
In this comment gave a small code snipped that prints the structure of the individual financial statements into the console. 
https://github.com/manusimidt/py-xbrl/issues/21#issuecomment-849371513



For more information see
#20 and #21

**Reply by jeffmarshall** [reply]
Thank you!
